setup.py

```diff
@@ -66,53 +85,83 @@ def run_tests(self):
+        TRANSLATE_SUFFIX = {"Windows": ".dll", "Darwin": ".dylib", "Linux": ".so"}
+        AVAILABLE_RELEASES = {
+            "Windows": ["win32", "x64"],
+            "Darwin": ["arm64", "x86_64"],
+            "Linux": ["aarch64", "x86_64"],
+        }
+        SUFFIX = TRANSLATE_SUFFIX[CURRENT_OS]
 
-        if not to_build:
-            # Build the extension as usual
-            BuildExtCommand.build_extension(self, ext)
+        if os.path.isdir(LIBDDWAF_DOWNLOAD_DIR) and len(os.listdir(LIBDDWAF_DOWNLOAD_DIR)):
             return
 
-        try:
-            cmake_command = os.environ.get("CMAKE_COMMAND", "cmake")
-            build_type = "RelWithDebInfo" if DEBUG_COMPILE else "Release"
-            opts = ["-DCMAKE_BUILD_TYPE={}".format(build_type)]
-            if platform.system() == "Windows":
-                opts.extend(["-A", "x64" if platform.architecture()[0] == "64bit" else "Win32"])
-            else:
-                opts.extend(["-G", "Ninja"])
-                ninja_command = os.environ.get("NINJA_COMMAND", "")
-                if ninja_command:
-                    opts.append("-DCMAKE_MAKE_PROGRAM={}".format(ninja_command))
-
-            for source_dir in to_build:
-                try:
-                    build_dir = tempfile.mkdtemp()
-                    subprocess.check_call([cmake_command, "-S", source_dir, "-B", build_dir] + opts)
-                    subprocess.check_call([cmake_command, "--build", build_dir, "--config", build_type])
-                finally:
-                    if not DEBUG_COMPILE:
-                        shutil.rmtree(build_dir, ignore_errors=True)
-
-            BuildExtCommand.build_extension(self, ext)
-        except Exception as e:
-            print('WARNING: building extension "%s" failed: %s' % (ext.name, e))
-            raise
+        if not os.path.isdir(LIBDDWAF_DOWNLOAD_DIR):
+            os.makedirs(LIBDDWAF_DOWNLOAD_DIR)
+
+        build_platform = get_build_platform()
+        for arch in AVAILABLE_RELEASES[CURRENT_OS]:
+            if CURRENT_OS == "Darwin" and build_platform.endswith("x86_64") and arch == "arm64":
+                continue
+
+            arch_dir = os.path.join(LIBDDWAF_DOWNLOAD_DIR, arch)
+
+            if os.path.isdir(arch_dir):
+                continue
+
+            ddwaf_archive_dir = "libddwaf-1.6.0-%s-%s" % (CURRENT_OS.lower(), arch)
+            ddwaf_archive_name = ddwaf_archive_dir + ".tar.gz"
+
+            ddwaf_download_address = (
+                "https://github.com/DataDog/libddwaf/releases/download/1.6.0/%s" % ddwaf_archive_name
+            )
+
+            try:
+                filename, http_response = urlretrieve(ddwaf_download_address, ddwaf_archive_name)
+                print(filename)
+            except HTTPError as e:
+                print("No archive found for dynamic library ddwaf : " + ddwaf_archive_dir)
+                raise e
+
+            with tarfile.open(filename, "r|gz", errorlevel=2) as tar:
```

Comment:
does it make sense to validate the checksum of the file we download?

this comment could be distracting, just the thought I had while reading this.

Reply:
The only way to validate the checksum right now would be to hard code it ourselves as it is not available from the github releases page or API. I agree it would be a nice addition, maybe in a subsequent PR?